src/lib.rs

```diff
@@ -14,10 +14,37 @@
 pub use arbitrary;
 use once_cell::sync::OnceCell;
 
+/// Indicates whether the input should be kept in the corpus or rejected. This
+/// should be returned by your fuzz target. If your fuzz target does not return
+/// a value (i.e., returns `()`), then the input will be kept in the corpus.
+#[derive(Debug)]
+pub enum Corpus {
+    /// Keep the input in the corpus.
+    Keep,
+
+    /// Reject the input and do not keep it in the corpus.
+    Reject,
+}
+
+impl From<()> for Corpus {
+    fn from(_: ()) -> Self {
+        Self::Keep
+    }
+}
+
+impl From<Corpus> for i32 {
+    fn from(value: Corpus) -> i32 {
+        match value {
+            Corpus::Keep => 0,
+            Corpus::Reject => -1,
+        }
+    }
+}
```

Comment:
Let's make this be something like a `Corpus::to_libfuzzer_code` method that is `#[doc(hidden)]` (but still `pub` so that is usable in the macro expanded code that is outside this crate) instead of a public trait impl, since this is only for internal use and shouldn't be exposed to users.

Reply:
Ayup, done!